app/src/main/java/org/literacyapp/authentication/MergeThread.java

```diff
@@ -134,7 +134,12 @@ private synchronized void findSimilarStudentsUsingMeanFeatureVector(PreProcessor
                 Student recognizedStudent = recognitionThread.getStudent();
                 if (recognizedStudent != null){
                     Log.i(getClass().getName(), "findSimilarStudentsUsingMeanFeatureVector: The student " + student.getUniqueId() + " has been recognized as " + recognizedStudent.getUniqueId());
-                    mergeSimilarStudents(student, recognizedStudent);
+                    if (recognizedStudent.getUniqueId().equals(student.getUniqueId())){
```

Comment:
To avoid performing this test, perhaps fetch only Students with a different id from the database instead of using `studentImageCollectionEventDao.loadAll()`?